flixel/graphics/frames/FlxAtlasFrames.hx

```diff
@@ -414,24 +412,158 @@ class FlxAtlasFrames extends FlxFramesCollection
 	}
 }
 
-typedef TexturePackerObject =
+typedef AtlasBase<T> = { frames:T }
+
+typedef AseAtlasBase<T> = AtlasBase<T> &
 {
-	frames:Dynamic
+	var meta:AseAtlasMeta;
+}
+
+typedef AseAtlasArray = AseAtlasBase<Array<AseAtlasFrame>>;
+typedef TexturePackerAtlasArray = AtlasBase<Array<AseAtlasFrame>>;
+
+abstract Hash<T>(Dynamic)
+{
+	public inline function keyValueIterator():KeyValueIterator<String, T>
+	{
+		var keys = Reflect.fields(this).iterator();
+		return
+		{
+			hasNext: keys.hasNext,
+			next: ()->
+			{
+				final key = keys.next();
+				return { key:key, value:Reflect.field(this, key)};
+			}
+		};
+	}
+}
+
+typedef AseAtlasHash = AseAtlasBase<Hash<AseAtlasFrame>>;
+typedef TexturePackerAtlasHash = AtlasBase<Hash<TexturePackerAtlasFrame>>;
+
+typedef HashOrArray<T> = flixel.util.typeLimit.OneOfTwo<Hash<T>, Array<T>>;
+
+typedef AseAtlas = AseAtlasBase<HashOrArray<AseAtlasFrame>>;
+typedef TexturePackerAtlas = AtlasBase<HashOrArray<TexturePackerAtlasFrame>>;
+
+typedef AseAtlasMeta =
+{
+	var app:String;
+	var version:String;
+	var image:String;
+	var format:String;
+	var size:AtlasSize;
+	var scale:String;
+	var frameTags:Array<AseAtlasTag>;
+	var slices:Array<AseAtlasSlice>;
+	var layers:Array<AseAtlasLayer>;
+}
+
+typedef AseAtlasTag = { name: String, from:Int, to:Int, direction:String };
+
+typedef AseAtlasSlice =
+{
+	var name:String;
+	var color:String;
+	
+	/**
+	 * Info of at what frames the slice changes size
+	 */
+	var keys: Array<AseAtlasSliceKey>;
 }
 
-typedef TexturePackerFrameRect =
+typedef AseAtlasLayer =
 {
-	x:Float,
-	y:Float,
-	w:Float,
-	h:Float
+	var name:String;
+	
+	/**
+	 * The name of the parent layer
+	 */
+	@:optional var group:String;
+	
+	/**
+	 * Ranges from 0 to 255
+	 */
+	@:optional var opacity:Int;
+	
+	/**
+	 * 
+	 */
+	@:optional var blendMode:AseBlendMode;
```

Comment:
It looks like `AseAtlasLayer` can also have user data attached to it. Set in-app by right clicking a cel -> `Cel Properties` -> Clicking the notepad icon

The data type appears to be of this format:
```
@:optional var cels:Array<AseCelData>;
```

```
typedef AseCelData = {
    frame:Int,
    data:String,
}
```

Reply:
thanks! I''ll add this, btw, I've noticed that color is not included if set to #000000